api/sso/routes.py

```diff
@@ -147,6 +153,11 @@ def _build_msal_app(cache=None, authority=None):
             token_cache=cache,
         )
 
+    @staticmethod
+    def _get_hostname_from_url(url):
```

Comment:
```suggestion
    def _get_origin_from_url(url):
```

nit: I think origin is the combination of protocol+hostname+port, so maybe that's more accurate than saying `hostname` here.